### PR TITLE
CI: Enable checkcommits

### LINKED ISSUE #16: Enable checkcommits
Enable [checkcommits](https://github.com/clearcontainers/tests/tree/master/cmd/checkcommits) which will requires that all PRs have:

- a `Signed-off-by:` line.
- a `Fixes #XXX` comment where `XXX` is a github issue number.
- Requires all commits to have a "sub-system" followed by a colon, followed by a brief description.
- Requires a blank line after the subsystem.
- Requires all commits to have a fuller description after the sub-system that will contain the `Fixes ###` and `Signed-off-by:` comments.

### PR DESCRIPTION
**Pull request by jodh-intel**
Require all PRs to contain a "Fixes #XXX" comment and a
"Signed-off-by:" comment.

Fixes #16.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: .ci/setup.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+set -e
+
+test_repo="github.com/clearcontainers/tests"
+
+# Clone Tests repository.
+go get "$test_repo"
+
+test_repo_dir="${GOPATH}/src/${test_repo}"
+
+# Check the commits in the branch
```
Comment (grahamwhaley) [inline .ci/setup.sh]: If we are having the same `checkcommits` config across all the repos in the project, do you think we can re-factor this call into a single script that lives in the https://github.com/kata-containers/tests repo.
Even if for now that script then grabs `checkcommits` itself from https://github.com/clearcontainers/tests whilst we work out if we import that wholesale or not into our kata tests repo?

> note - this stems from the long standing idea that we needed to factorise the go static checks out of their individual repos and into a single file under Clear Containers - something we never quite got around to doing.
> let's not end up in that place again :-)

Comment (grahamwhaley) [reply .ci/setup.sh]: I guess to be even more explicit. If this script did something like:
```
test_repo="github.com/clearcontainers/tests"
# Clone Tests repository.
go get "$test_repo"
# Run the 'checkers'
${tests_repo}/.ci/run_all_static_checks.sh
```

and then we can formalise and keep all the static checking (checkmetrics, static go linters etc.) in one place.

Obviously, this same comment goes for all the other repos under kata where you have the same PRs sat :-)

Comment (jodh-intel) [reply .ci/setup.sh]: I like the idea. Respinning bytes... :)

Comment (jodh-intel) [reply .ci/setup.sh]: Hi @grahamwhaley - how about https://github.com/kata-containers/tests/pull/7?

**Comment by jodh-intel** [comment]
See https://github.com/kata-containers/tests/pull/7.

**Comment by grahamwhaley** [comment]
looked good to me - left feedback over there...

**Comment by jodh-intel** [comment]
Grrrr, I've been hit by the dreaded "Travis bug" (https://github.com/travis-ci/travis-ci/issues/7830). Closing and re-opening to force Travis to reset...

**Pull request closed without merging**